### ISSUE TITLE
Safari support?

### ISSUE THREAD

**Issue by bdougherty** [ClearURLs/Addon#92]
Now that Safari supports the WebExtension API, do you think it would be possible to offer a Safari version? Happy to help if I can.

**Comment by securingmom** [comment]
"Dad, stop using Safari" is always my answer to similar questions.  Yes, I am using a Macbook Pro right now so this isn't anti-BSD.

**Comment by finngaida** [comment]
If you have Xcode installed you can create a Safari extension quite easily yourself by running 
```
xcrun safari-web-extension-converter Addon
```

There are some more detailed instructions here https://bartsolutions.github.io/2020/11/20/safari-extension/

**Comment by adammenges** [comment]
Great idea. Would be sweet to get this in and listed.

**Comment by Joilence** [comment]
Unfortunately, the current latest Safari 14.1 does not support `WebRequestBlocking`. It seems to be not possible to convert. Saw this issue in other extensions that try to move to Safari, too.

**Comment by KevinRoebert** [comment]
> Unfortunately, the current latest Safari 14.1 does not support `WebRequestBlocking`. It seems to be not possible to convert. Saw this issue in other extensions that try to move to Safari, too.

Yes, that's right. Unfortunately, the `WebRequestBlocking` is the core API required by ClearURLs. We have to wait until Apple will implement this.

**Comment by rakleed** [comment]
@KevinRoebert [Safari 15](https://developer.apple.com/documentation/safari-release-notes/safari-15-beta-release-notes) will add support for [declarativeNetRequest](https://developer.apple.com/documentation/safariservices/safari_web_extensions/blocking_content_with_your_safari_web_extension). Can this help you?

**Comment by KevinRoebert** [comment]
> @KevinRoebert [Safari 15](https://developer.apple.com/documentation/safari-release-notes/safari-15-beta-release-notes) will add support for [declarativeNetRequest](https://developer.apple.com/documentation/safariservices/safari_web_extensions/blocking_content_with_your_safari_web_extension). Can this help you?

Unfortunately, this does not help. This is Chrome's `webRequestBlocking` replacement for the WebExtentensions v3 API. This only allows you to create restricted filter lists that no longer allow individual treatment. So you can no longer remove (or add) elements from a URL, just block a request entirely. However, this is useless for ClearURLs (and also many other add-ons like uBlock or NoScript). We don't want to discard an entire request due to a (restricted) filter but process this request to remove possible "harmful" content from it.

This topic was also covered on Reddit by gorhill: https://www.reddit.com/r/uBlockOrigin/comments/hdz0bo/will_ublock_origin_back_to_macos_big_sur/fvoc7wk/

This may also make it impossible to use addons like ClearURLs, uBlock, or NoScript in future versions of Chrome.

We can only hope that Apple will implement the `webRequestBlocking` API. Until that happens, I can't provide the addon for Safari.

Another problem would be the exorbitant costs for developers. 99€ per year to deploy a free!!! addon. But that would be a problem that could be addressed once/if Apple provides the necessary API.

**Comment by KevinRoebert** [comment]
> Any updates on this?

No, Safari still lacks the required API, and it seems unlikely that they will implement it now that the API has been removed from Manifest V3.
